### PR TITLE
Adapt WKBotsFeeder plugin to new buildbot API

### PR DESCRIPTION
**Pull request by dpino**
Recently build.webkit.org has updated to buildbot 2.10, which uses a new JSON API (See: http://docs.buildbot.net/current/developer/rest.html).

This PR is WIP. I'm testing these changes locally, but I need to wait for results to verify it's working correctly.

### CODE REVIEW

**Comment by dpino** [comment]
Forgot to mention that besides these code changes, it's also necessary to change the URLs in the configuration file of the plugin. 
The new URLs are:

```python
plugin_wk["settings"] = {
   "last_buildjob_url_schema": "https://build.webkit.org/#/builders/%d/builds/%d",
   "builds_url_schema": "https://build.webkit.org/api/v2/builders/%d/builds?complete=true&order=-number&limit=1",
}
```

**Comment by dpino** [comment]
This has been tested and it's working:

```
GTK-Linux-64-bit-Release-Ubuntu-LTS-Build (772 ): success
GTK-Linux-64-bit-Debug-Build (539 ): success
WPE-Linux-64-bit-Debug-Build (586 ): success
WPE-Linux-64-bit-Release-Build (726 ): success
GTK-Linux-64-bit-Release-Debian-Stable-Build (779 ): success
GTK-Linux-64-bit-Release-Build (693 ): success
```

File: matrixbot/plugins/wkbotsfeeder.py [hunk 1]
```diff
@@ -128,3 +136,41 @@ def command(self, sender, room_id, body, handler):
...
+    print("test_async: ")
+    logging.basicConfig(level = logging.DEBUG)
+    plugin.lasttime = 0
+    plugin.period = 0
+    plugin.async()
+    print("")
+    print("Ok")
+
+def test_can_fetch_last_build(plugin):
+    put("test_can_fetch_last_build: ")
```
Comment (psaavedra) [inline matrixbot/plugins/wkbotsfeeder.py:169]: What is the idea behind  use `sys.stdout.write` instead of a simple `print()`

**Review by psaavedra — CHANGES_REQUESTED** [review]

File: matrixbot/plugins/wkbotsfeeder.py (hunk 1 above)
Comment (dpino) [reply matrixbot/plugins/wkbotsfeeder.py:169]: `print()` always prints a newline character. `sys.stdout.write()` prints without a new character.

File: matrixbot/utils.py [hunk 2]
```diff
@@ -108,3 +108,14 @@ def get_command_alias(message, settings):
...
+
+class MockBot:
+    def __init__(self):
+        pass
+
+    def get_real_room_id(self, room_id):
+        return room_id or 0
+
+    def send_html(self, room_id, message, **kwargs):
+        sys.stdout.write("Room #{}: {}".format(room_id, message))
```
Comment (psaavedra) [inline matrixbot/utils.py:121]: Let's use "put" here as alias of "sys.stdout.write" too.

File: matrixbot/plugins/wkbotsfeeder.py [hunk 3]
```diff
@@ -128,3 +136,41 @@ def command(self, sender, room_id, body, handler):
...
+    builder = plugin.settings['builders']["GTK-Linux-64-bit-Release-Ubuntu-LTS-Build"]
+    build = plugin.get_last_build(builder)
+    assert(build)
+    print("Ok")
```
Comment (psaavedra) [inline matrixbot/plugins/wkbotsfeeder.py:173]: Nit: Let's keep using the same method to do the output. WDYT `put("OK\n")`?

File: matrixbot/utils.py [hunk 4]
```diff
@@ -108,3 +108,14 @@ def get_command_alias(message, settings):
 def get_aliases(settings):
     res = copy.copy(settings["aliases"])
     return res
+
+
+class MockBot:
```
Comment (psaavedra) [inline matrixbot/utils.py:113]: Nice to have a mock server to test the plugin.

File: matrixbot/plugins/wkbotsfeeder.py (hunk 3 above)
Comment (dpino) [reply matrixbot/plugins/wkbotsfeeder.py:173]: I borrowed this `puts` and `print`terminology from Ruby. In Ruby it works the opposite way, `puts` prints a newline while `print` doesn't. I think is legit to use either `puts` or `print` depending on whether is necessary to write a newline (likewise we won't change all the print statements to puts with a newline.

File: matrixbot/utils.py (hunk 4 above)
Comment (dpino) [reply matrixbot/utils.py:115]: Actually is very convenient to have these selftest methods that let us test a plugin very quickly without having to deploy it.

**Review by psaavedra — APPROVED** [review]

**Pull request merged**